MetadataExtractor/TagDescriptor.cs

```diff
@@ -221,8 +221,7 @@ public TagDescriptor(T directory)
 
             try
             {
-                return Encoding.UTF8
-                    .GetString(values, 0, values.Length)
+                return Encoding.UTF8.GetString(values)
                     .Trim('\0', ' ', '\r', '\n', '\t');
```

Comment:
I was wondering if we could trim a span rather than a string, but I don't think it's safe to trim these bytes in all UTF8 strings, and that trimming characters is better. It would be possible to use the `Encoding` to populate a `Span<char>` and trim that, but I don't think it's worth it.

My plan is to look at some traces and take a data-led approach to the next wave of optimisations. There's too much code to go through.

Reply:
This is also a good case for the ArrayPool, where we trim the buffer, before materializing the string.

Reply:
We should also try to eliminate some of these silently allocated arrays when using functions that accept a params T[] array (as is the case above).